### PR TITLE
Added reheadSQ script and compilation instructions

### PR DESCRIPTION
**Pull request by drjsanger**
@keiranmraine Please check

### CODE REVIEW

**Comment by keiranmraine** [comment]
@drjsanger , right back at you.  Tested `bwa_mem.pl` with the dict file and got a nice shiny new BAM with new header fields.

FYI the additional compile deps prevent us testing the C code with pre-release, we were getting roung the `HTSLIB` dep by testing for an env variable, could this be done for the bigwig lib too?

**Comment by drjsanger** [comment]
@keiranmraine Updated prerelease.sh and makefile for C to use $LIBBIGWIG presence as a decision for testing the c code. I've had a quick test and it works. Please test.

**Comment by keiranmraine** [comment]
@drjsanger , for `LIBBIGWIG` the code is expected a fully deployed library area rather than a workspace following a basic `make`.  Is this expected?

````
ls libBigWig/
bigWig.h    bwRead.o	 bwStats.o     bwValues.h     bwWrite.c     docs      io.h     libBigWig.a   Makefile
bwCommon.h  bwRead.pico  bwStats.pico  bwValues.o     bwWrite.o     Doxyfile  io.o     libBigWig.so  README.md
bwRead.c    bwStats.c	 bwValues.c    bwValues.pico  bwWrite.pico  io.c      io.pico  LICENSE	     test
````

`HTSLIB` works of a compiled workspace, rather than a deployed version so I was expecting that this would be the same.

My working:

````
git clone git@github.com:dpryan79/libBigWig.git
cd libBigWig
make
export LIBBIGWIG=`pwd`
cd ../PCAP-core
./prerelease.sh
...
remove tmp hts location
rm -rf ./htslib_tmp
gcc -O3 -DVERSION='"2.1.3"' -g -I?/usr/local//include -I/nfs/users/nfs_k/kr2/GitHub/libBigWig/include -Wall ./bwcat.c -L?/usr/local//lib -L/nfs/users/nfs_k/kr2/GitHub/libBigWig/lib -lBigWig -lz -lcurl -lm -lgnutls -ltasn1 -lhogweed -lnettle -lgmp -lp11-kit -o ../bin/bwcat
./bwcat.c:7:20: fatal error: bigWig.h: No such file or directory
compilation terminated.
make: *** [../bin/bwcat] Error 1
make: Leaving directory `/nfs/users/nfs_k/kr2/GitHub/PCAP-core/c'
````

**Comment by keiranmraine** [comment]
@drjsanger problem with biobambam2 required update.

**Comment by drjsanger** [comment]
@keiranmraine I've corrected the LIBBIGWIG to function as you expected. Please double check.

**Comment by keiranmraine** [comment]
@drjsanger it progresses beyond the libBigWig step now, but I think we'll need to re-think as it needs quite a lot of pre-requisites to function which make it quite unlikely that anyone else will set this up properly:

````
/usr/bin/ld: cannot find -lhogweed
/usr/bin/ld: cannot find -lnettle
````

File: c/bwcat.c [hunk 1]
```diff
@@ -93,15 +100,49 @@ void setup_options(int argc, char *argv[]){
...
+  char **ignore_list = NULL;
+  char *copy = malloc(sizeof(char) * strlen(contigs)+1);
+  strcpy(copy,contigs);
+  char *tag = strtok(copy,",");
+  ign_count=0;
+  while(tag != NULL){
+    ign_count++;
+    tag = strtok(NULL,",");
+  }
+  ignore_list = malloc(sizeof(char *) * count);
```
Comment (keiranmraine) [inline c/bwcat.c]: @drjsanger , doesn't compile, should be:

````
ignore_list = malloc(sizeof(char *) * ign_count);
````

File: c/bwcat.c [hunk 2]
```diff
@@ -188,6 +229,8 @@ int main(int argc, char *argv[]){
 
   int j=0;
   for(j=0;j<chromList->nKeys;j++){//Iterate through each contig and access file.
+
+    if(ignore != NULL && check_ignore(ignore,ign_count,chromList->chrom[j])) continue;
```
Comment (keiranmraine) [inline c/bwcat.c]: @drjsanger , other compiler errors.  Not sure setup.sh is working at all anymore for the bigwig stuff:

````
./bwcat.c:233:5: warning: passing argument 1 of 'check_ignore' from incompatible pointer type [enabled by default]
     if(ignore != NULL && check_ignore(ignore,ign_count,chromList->chrom[j])) continue;
     ^
./bwcat.c:134:5: note: expected 'const char **' but argument is of type 'char **'
 int check_ignore(const char **ignore_list, const int ign_size, const char *contig){
     ^
/tmp/cc5MuPLg.o: In function `main':
/tmp/downloads/PCAP-core/c/./bwcat.c:314: undefined reference to `bwClose'
/tmp/downloads/PCAP-core/c/./bwcat.c:317: undefined reference to `bwClose'
/tmp/downloads/PCAP-core/c/./bwcat.c:318: undefined reference to `bwCleanup'
/tmp/downloads/PCAP-core/c/./bwcat.c:224: undefined reference to `bwOpen'
/tmp/downloads/PCAP-core/c/./bwcat.c:297: undefined reference to `bwClose'
/tmp/downloads/PCAP-core/c/./bwcat.c:298: undefined reference to `bwCleanup'
/tmp/downloads/PCAP-core/c/./bwcat.c:237: undefined reference to `bwInit'
/tmp/downloads/PCAP-core/c/./bwcat.c:245: undefined reference to `bwOpen'
/tmp/downloads/PCAP-core/c/./bwcat.c:269: undefined reference to `bwGetOverlappingIntervals'
/tmp/downloads/PCAP-core/c/./bwcat.c:286: undefined reference to `bwAddIntervals'
/tmp/downloads/PCAP-core/c/./bwcat.c:292: undefined reference to `bwDestroyOverlappingIntervals'
/tmp/downloads/PCAP-core/c/./bwcat.c:293: undefined reference to `bwClose'
/tmp/downloads/PCAP-core/c/./bwcat.c:314: undefined reference to `bwClose'
/tmp/downloads/PCAP-core/c/./bwcat.c:315: undefined reference to `bwDestroyOverlappingIntervals'
/tmp/downloads/PCAP-core/c/./bwcat.c:314: undefined reference to `bwClose'
/tmp/downloads/PCAP-core/c/./bwcat.c:253: undefined reference to `bwCreateHdr'
/tmp/downloads/PCAP-core/c/./bwcat.c:257: undefined reference to `bwCreateChromList'
/tmp/downloads/PCAP-core/c/./bwcat.c:262: undefined reference to `bwWriteHdr'
collect2: error: ld returned 1 exit status
make: *** [../bin/bwcat] Error 1
make: Leaving directory `/tmp/downloads/PCAP-core/c'
````

**Comment by drjsanger** [comment]
I've fixed bwcat to the point it will compile if pointed to the correct location for libBigWig. So should just be makefile alterations

**Comment by keiranmraine** [comment]
@drjsanger trying to build on an AWS instance with nettle/hogweed installed it fails with:

````
$ ./setup.sh /opt/wtsi-cgp
...
RUNNING: ./c_tests/03_bam_stats_calcs_tests
ALL TESTS PASSED
Tests run: 1
c_tests/03_bam_stats_calcs_tests PASS

remove tmp hts location
rm -rf ./htslib_tmp
gcc -O3 -DVERSION='"2.1.3"' -g -I/opt/wtsi-cgp/include -I -Wall ./bwcat.c -L/opt/wtsi-cgp/lib -L -lBigWig -lz -lcurl -lm -lgnutls -ltasn1 -lhogweed -lnettle -lgmp -lp11-kit -o ../bin/bwcat
/tmp/ccwrNx2E.o: In function `main':
/tmp/downloads/distro/c/./bwcat.c:314: undefined reference to `bwClose'
/tmp/downloads/distro/c/./bwcat.c:317: undefined reference to `bwClose'
/tmp/downloads/distro/c/./bwcat.c:318: undefined reference to `bwCleanup'
/tmp/downloads/distro/c/./bwcat.c:224: undefined reference to `bwOpen'
/tmp/downloads/distro/c/./bwcat.c:297: undefined reference to `bwClose'
/tmp/downloads/distro/c/./bwcat.c:298: undefined reference to `bwCleanup'
/tmp/downloads/distro/c/./bwcat.c:237: undefined reference to `bwInit'
/tmp/downloads/distro/c/./bwcat.c:245: undefined reference to `bwOpen'
/tmp/downloads/distro/c/./bwcat.c:269: undefined reference to `bwGetOverlappingIntervals'
/tmp/downloads/distro/c/./bwcat.c:286: undefined reference to `bwAddIntervals'
/tmp/downloads/distro/c/./bwcat.c:292: undefined reference to `bwDestroyOverlappingIntervals'
/tmp/downloads/distro/c/./bwcat.c:293: undefined reference to `bwClose'
/tmp/downloads/distro/c/./bwcat.c:314: undefined reference to `bwClose'
/tmp/downloads/distro/c/./bwcat.c:315: undefined reference to `bwDestroyOverlappingIntervals'
/tmp/downloads/distro/c/./bwcat.c:314: undefined reference to `bwClose'
/tmp/downloads/distro/c/./bwcat.c:253: undefined reference to `bwCreateHdr'
/tmp/downloads/distro/c/./bwcat.c:257: undefined reference to `bwCreateChromList'
/tmp/downloads/distro/c/./bwcat.c:262: undefined reference to `bwWriteHdr'
collect2: error: ld returned 1 exit status
make: *** [../bin/bwcat] Error 1
make: Leaving directory `/tmp/downloads/distro/c'
````

But it looks like liBigWig files have been installed correctly:

````
$ ls /opt/wtsi-cgp/include/
bigWig.h  bwCommon.h  bwValues.h  io.h
$ ls /opt/wtsi-cgp/lib/    
libBigWig.a  perl5

````

**Comment by keiranmraine** [comment]
@drjsanger , `setup.sh` runs cleanly on fresh ubuntu instance.  You want this merging now?

**Pull request merged**